### PR TITLE
[ECF-1-497] Lead provider dashboard layout

### PR DESCRIPTION
**Pull request by amazimbe**
### Context
The dashboard layout for lead providers has changed from a two column / block layout to a simple heading/paragraph layout

Also, it's using the correct navigation component now (after checking with Alaric)—and the tabs component has been renamed to subnav to prevent this from happening again.

### Review Checks
- [x] All pages have automated accessibility checks via cypress
- [x] All pages have visual tests via cypress + percy
- [x] All `School` queries are correctly scoped - `eligible` when they need to be

### How can I view this in a review app?
Login as lead-provider@example.com and check the dashboard.

Screenshot 2021-04-20 at 18 29 03

### CODE REVIEW

**Comment by amazimbe** [comment]
[Update Lead Provider Dashboard](https://trello.com/c/05POg6Ow/497-update-lead-provider-dashboard)

**Comment by amazimbe** [comment]
[Update Lead Provider Dashboard](https://trello.com/c/05POg6Ow/497-update-lead-provider-dashboard)

File: spec/cypress/integration/Dashboard.feature [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Feature: Dashboard page
+  Background:
+    Given cohort was created with start_year "2021"
+  Scenario: Visiting the dashboard
+    Given I am logged in as a "lead_provider"
+    Then I should be on "dashboard" page
+    And "page body" should contain "Confirm your schools"
+    And "page body" should contain "Check your schools"
+    And the page should be accessible
```
Comment (kicferk1) [inline spec/cypress/integration/Dashboard.feature:9]: And a Percy shapshot?

File: spec/cypress/support/step_definitions/common-navigation.js [hunk 2]
```diff
@@ -19,6 +19,7 @@ const pagePaths = {
   start: "/",
   privacy: "/privacy-policy",
   accessibility: "/accessibility-statement",
+  dashboard: "/dashboard",
```
Comment (kicferk1) [inline spec/cypress/support/step_definitions/common-navigation.js:22]: Ech, I anticipate conflicts with https://github.com/DFE-Digital/early-careers-framework/pull/268 :/

File: spec/requests/dashboard_spec.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe "Dashboard", type: :request do
+  let(:lead_provider) { create(:user, :lead_provider) }
+
+  before do
+    sign_in lead_provider
+    Cohort.create!(start_year: "2021")
```
Comment (BroiSatse) [inline spec/requests/dashboard_spec.rb:10]: I think this is a bit dangerous here. 2021 seems to be hardcoded quite a lot in the test and we'll spend the first week(s) after Christmas fixing test failures. Could we have `create :cohort, :current` instead or use time travel?

**Review by kicferk1 — APPROVED** [review]

File: app/views/dashboard/_navigation.html.erb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<%= render TabsComponent.new do |component| %>
+  <%= component.tab(path: dashboard_path) do %>
+    <strong>Overview</strong>
+  <% end %>
+  <%= component.tab(path: admin_supplier_users_path) do %>
```
Comment (kicferk1) [inline app/views/dashboard/_navigation.html.erb:5]: Uhh, does that path look wrong to you too?

Comment (amazimbe) [reply app/views/dashboard/_navigation.html.erb:5]: It does but then it doesn't really matter. I need to rebase once the report schools pr has been merged and that will be corrected then. Or perhaps I can copy your routes into mine and worry about the conflicts later

File: app/views/dashboard/show.html.erb [hunk 5]
```diff
@@ -2,22 +2,17 @@
...
+        <%= render partial: "navigation" %>
+        <h1 class="govuk-heading-xl"><%= @lead_provider.name %></h1>
+        <h2 class="govuk-heading-m">
+          <!-- TODO Fix after Ethan code merge for lead providers-->
+          <%= govuk_link_to "Confirm your schools", nil %>
+        </h2>
+        <p>Tell the Department for Education (DfE) which schools you have confirmed an agreement with.</p>
+        <h2 class="govuk-heading-m">
+          <%= govuk_link_to "Check your schools", lead_providers_your_schools_path(selected_cohort_id: Cohort.current.id) %>
+        </h2>
```
Comment (ethanmills) [inline app/views/dashboard/show.html.erb:14]: I think the ticket says not to have this?

Comment (amazimbe) [reply app/views/dashboard/show.html.erb:14]: That's right

File: app/views/dashboard/_navigation.html.erb [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<%= render TabsComponent.new do |component| %>
+  <%= component.tab(path: dashboard_path) do %>
+    <strong>Overview</strong>
+  <% end %>
+  <%= component.tab(path: start_lead_providers_report_schools_path) do %>
+    <strong>Schools</strong>
+  <% end %>
+<% end %>
```
Comment (kicferk1) [inline app/views/dashboard/_navigation.html.erb:8]: I might have added the same navigation inline in the report page. Could you reuse this version there?

**Review by kicferk1 — APPROVED** [review]

File: app/views/dashboard/_navigation.html.erb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+<%= render TabsComponent.new do |component| %>
+  <%= component.tab(path: dashboard_path) do %>
+    <strong>Overview</strong>
```
Comment (callumacrae) [inline app/views/dashboard/_navigation.html.erb:3]: Why are these being styled bold here in this way? They should either be styled bold for all sub navigation across the site using CSS, or they shouldn't be bold.

**Review by callumacrae — CHANGES_REQUESTED** [review]

File: app/views/dashboard/_navigation.html.erb (hunk 7 above)
Comment (amazimbe) [reply app/views/dashboard/_navigation.html.erb:3]: All the designs I have seen had them in bold. I agree that should be done in css though

Comment (callumacrae) [reply app/views/dashboard/_navigation.html.erb:3]: Oh, right, I just checked the design and I understand now.

In that case, I think this is using the wrong component - the wireframes for the admin flow have the same thing and it looks like this on the real site:

image

Comment (amazimbe) [reply app/views/dashboard/_navigation.html.erb:3]: @callumacrae from what Alaric said there's just one component that does this but it needs to support bold font if at the top and normal font if inside the page. The component this is based on is  https://hmcts-design-system.herokuapp.com/components/sub-navigation . What I have done is correct and if I can't see any benefit from styling this in css in this case.

Comment (callumacrae) [reply app/views/dashboard/_navigation.html.erb:3]: image

For reference, this is what the page I just posted a screenshot of above looked like in the prototypes.

In our implementation currently, there's more than one component that does this! Per the HMCTS design system, that component is for _sub_ navigation, but we're dealing with primary navigation here (I don't know why it was renamed to Tabs when it was copied into this repo - it shouldn't have been).

If we're going to go with the design in this PR, we should still use the primary navigation component, but we should change it where it's already used, too. We 100% shouldn't have inconsistent navigation between pages!

File: app/webpacker/styles/subnav.scss [hunk 8]
```diff
@@ -0,0 +1,118 @@
+/* ==========================================================================
```
Comment (callumacrae) [inline app/webpacker/styles/subnav.scss:1]: This is just page-tabs.scss but actually in SCSS format

**Review by callumacrae — APPROVED** [review]

File: db/schema.rb [hunk 9]
```diff
@@ -205,6 +205,10 @@
     t.uuid "school_id", null: false
     t.uuid "lead_provider_id", null: false
     t.uuid "cohort_id", null: false
+    t.string "status", default: "pending", null: false
+    t.string "reason_for_rejection"
+    t.datetime "accepted_at"
+    t.datetime "rejected_at"
```
Comment (kicferk1) [inline db/schema.rb:211]: That seems... surprising

Comment (callumacrae) [reply db/schema.rb:211]: dammit i'll revert this

**Review by kicferk1 — APPROVED** [review]

**Review by ethanmills — APPROVED** [review]
Super minor thing, but I think primary nav is supposed to have font weight 700 @callumacrae

File: app/views/admin/schools/_navigation.html.erb [hunk 10]
```diff
@@ -1,11 +1,11 @@
-<%= render TabsComponent.new do |component| %>
-  <%= component.tab(path: admin_school_path(@school)) do %>
+<%= render SubnavComponent.new do |component| %>
+  <%= component.nav_item(path: admin_school_path(@school)) do %>
     <strong>Overview</strong>
```
Comment (ethanmills) [inline app/views/admin/schools/_navigation.html.erb:3]:
```suggestion
    Overview
```

File: app/views/admin/schools/_navigation.html.erb [hunk 11]
```diff
@@ -1,11 +1,11 @@
...
   <% end %>
-  <%= component.tab(path: "#") do %>
+  <%= component.nav_item(path: "#") do %>
     <strong>Participants</strong>
```
Comment (ethanmills) [inline app/views/admin/schools/_navigation.html.erb:6]:
```suggestion
    Participants
```

File: app/views/admin/schools/_navigation.html.erb [hunk 12]
```diff
@@ -1,11 +1,11 @@
...
   <% end %>
-  <%= component.tab(path: "#") do %>
+  <%= component.nav_item(path: "#") do %>
     <strong>Cohorts</strong>
```
Comment (ethanmills) [inline app/views/admin/schools/_navigation.html.erb:9]:
```suggestion
    Cohorts
```

**Review by ethanmills — COMMENTED** [review]
Sorry... One last thing?

File: app/views/admin/schools/_navigation.html.erb (hunk 10 above)
Comment (callumacrae) [reply app/views/admin/schools/_navigation.html.erb:3]: I thought I got rid of these!

Comment (callumacrae) [reply app/views/admin/schools/_navigation.html.erb:3]: oh wait this isn't part of this PR. i'll change anyway 😆

Comment (ethanmills) [reply app/views/admin/schools/_navigation.html.erb]: Thanks!

**Pull request merged**